### PR TITLE
wasm expose: stakepool funcs

### PR DESCRIPTION
**Pull request by boddumanohar**
### Version
- [x] Ensure to run `make getrev` to update gosdk version info.

### Changes
exposed
- lockStakePool
- unlockStakePool
- getstakePoolInfo
 

### Fixes
-

### Tests
Tasks to complete before merging PR:
- [x]  Ensure system tests are passing. If not [Run them manually](https://github.com/0chain/gosdk/actions/workflows/system_tests.yml) to check for any regressions :clipboard:
- [x]  Do any new system tests need added to test this change? do any existing system tests need updated? If so create a PR at [0chain/system_test](https://github.com/0chain/system_test)
- [x]  Merge your system tests PR to master AFTER merging this PR

### Associated PRs (Link as appropriate):
- blobber:
- 0chain:
- system_test:
- zboxcli:
- zwalletcli:
- Other: ...

### CODE REVIEW

File: wasmsdk/allocation.go [hunk 1]
```diff
@@ -214,6 +214,42 @@ func lockWritePool(allocID, tokens, fee string) (string, error) {
 	return hash, err
 }
 
+func lockStakePool(providerType int, tokens, providerID, fee string) (string, error) {
```
Comment (cnlangzi) [inline wasmsdk/allocation.go]: uint64 is supported on wasm, `ParseCoinStr` is unnecessary here.

**Review by cnlangzi — CHANGES_REQUESTED** [review]
some small fixes

File: wasmsdk/allocation.go (hunk 1 above)
Comment (boddumanohar) [reply wasmsdk/allocation.go]: fixed in 6cc0793

**Review by cnlangzi — APPROVED** [review]
LGTM

**Comment by service-0chain** [comment]
Manual system tests [success] with the following config
| config  | value |
| ------------- | ------------- |
| tests skipped  | ABILITY TO SKIP TESTS IS DISABLED. SKIP_TESTS OPTION IGNORED.  |
| system-tests  | kill_shudown_config  |
| gosdk  | staging  |
| zboxcli  | staging  |
| zwalletcli  | staging  |
| 0chain  | blobber_kill_shutdown  |
| blobber  | staging  |
| authorizer  | staging  |
| 0box  | staging  |
| 0block  | staging  |
| 0dns  | staging  |
| explorer  | staging  |
| 0proxy  | staging  |
| 0search  | staging  |

**Pull request merged**